sdk/servicebus/Microsoft.Azure.ServiceBus/src/Core/MessageReceiver.cs

```diff
@@ -879,6 +879,81 @@ public async Task<IList<Message>> PeekBySequenceNumberAsync(long fromSequenceNum
             return messages;
         }
 
+        public Task<int> BatchDeleteMessagesAsync(int messageCount)
+        {
+            return this.BatchDeleteMessagesAsync(messageCount, this.OperationTimeout);
+        }
+
+        internal async Task<int> BatchDeleteMessagesAsync(int messageCount, TimeSpan timeout)
+        {
+            MessagingEventSource.Log.MessageBatchDeleteStart(this.ClientId, timeout, messageCount);
+            bool isDiagnosticSourceEnabled = ServiceBusDiagnosticSource.IsEnabled();
+            Activity activity = isDiagnosticSourceEnabled ? this.diagnosticSource.BatchDeleteStart(timeout, messageCount) : null;
+            Task deleteTask = null;
+            int messagesDeleted = 0;
+
+            try
+            {
+                deleteTask = this.RetryPolicy.RunOperation(
+                    async () =>
+                    {
+                        messagesDeleted = await this.OnBatchDeleteMessagesAsync(messageCount, timeout).ConfigureAwait(false);
+                    }, this.OperationTimeout);
+
+                await deleteTask.ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                if (isDiagnosticSourceEnabled)
+                {
+                    this.diagnosticSource.ReportException(exception);
+                }
+
+                MessagingEventSource.Log.MessagePeekException(this.ClientId, exception);
+                throw;
+            }
+            finally
+            {
+                this.diagnosticSource.BatchDeleteStop(activity, timeout, messageCount, deleteTask?.Status, messagesDeleted);
```

Comment:
1. Should we check the isDaignosticSourceEnabled flag here before?
2. Why do we log the start and end with different logging providers?
3. There is the possibility that 'activity' could be null - is that ok?